// knigabel-client/src/styles/themes.js
export const lightTheme = {
  // Colors - Minimalist palette
  primary: '#2D3748',
  primaryDark: '#1A202C',
  primaryLight: '#4A5568',
  secondary: '#718096',
  secondaryDark: '#4A5568',
  accent: '#E2E8F0',
  accentDark: '#CBD5E1',

  // Status colors - Subtle
  success: '#48BB78',
  error: '#F56565',
  warning: '#ED8936',
  info: '#4299E1',

  // Background colors - Clean whites and grays
  background: '#FFFFFF',
  backgroundSolid: '#FFFFFF',
  backgroundSecondary: '#F7FAFC',
  backgroundTertiary: '#EDF2F7',

  // Text colors - High contrast but soft
  text: '#1A202C',
  textLight: '#4A5568',
  textLighter: '#718096',

  // Border colors - Subtle
  border: '#E2E8F0',
  borderLight: '#EDF2F7',

  // Card styles - Clean and minimal
  card: '#FFFFFF',
  cardShadow: '0 1px 3px rgba(0, 0, 0, 0.1), 0 1px 2px rgba(0, 0, 0, 0.06)',
  cardBorder: '#E2E8F0',

  // Navigation - Clean
  navBackground: '#FFFFFF',
  navActive: '#2D3748',

  // Gradients - Subtle
  gradientPrimary: 'linear-gradient(135deg, #2D3748, #1A202C)',
  gradientSecondary: 'linear-gradient(135deg, #718096, #4A5568)',
  gradientAccent: 'linear-gradient(135deg, #E2E8F0, #CBD5E1)',
  gradientCard: 'linear-gradient(135deg, #FFFFFF, #F7FAFC)',

  // Header - Minimalist
  headerBackground: 'linear-gradient(135deg, #FFFFFF 0%, #F7FAFC 100%)',
  headerText: '#1A202C',

  // Hero - Clean
  heroBackground: '#FFFFFF',
  heroCardBackground: 'rgba(255, 255, 255, 0.8)',

  // Shadows - Subtle
  shadow: '0 1px 3px rgba(0, 0, 0, 0.1)',
  shadowHover: '0 4px 6px rgba(0, 0, 0, 0.05)',
};

export const darkTheme = {
  // Colors
  primary: '#06B6D4',
  primaryDark: '#0891B2',
  primaryLight: '#22D3EE',
  secondary: '#10B981',
  secondaryDark: '#059669',
  accent: '#F59E0B',
  accentDark: '#D97706',

  // Status colors
  success: '#10B981',
  error: '#EF4444',
  warning: '#F59E0B',
  info: '#3B82F6',

  // Background colors
  background: '#0F172A',
  backgroundSolid: '#1E293B',
  backgroundSecondary: '#1E293B',
  backgroundTertiary: '#334155',

  // Text colors
  text: '#F8FAFC',
  textLight: '#CBD5E1',
  textLighter: '#94A3B8',

  // Border colors
  border: '#334155',
  borderLight: '#475569',

  // Card styles
  card: '#1E293B',
  cardShadow: '0 1px 3px rgba(0, 0, 0, 0.3)',
  cardBorder: '#334155',

  // Navigation
  navBackground: 'linear-gradient(135deg, #1E293B, #334155)',
  navActive: 'linear-gradient(135deg, #06B6D4, #0891B2)',

  // Gradients
  gradientPrimary: 'linear-gradient(135deg, #06B6D4, #0891B2)',
  gradientSecondary: 'linear-gradient(135deg, #10B981, #059669)',
  gradientAccent: 'linear-gradient(135deg, #F59E0B, #D97706)',
  gradientCard: 'linear-gradient(135deg, #1E293B, #334155)',

  // Header
  headerBackground: 'linear-gradient(135deg, #1e293b 0%, #334155 50%, #475569 100%)',
  headerText: '#F8FAFC',

  // Hero
  heroBackground: 'linear-gradient(135deg, #0f172a 0%, #1e293b 50%, #334155 100%)',
  heroCardBackground: 'linear-gradient(135deg, rgba(30,41,59,0.8), rgba(51,65,85,0.6))',

  // Shadows
  shadow: '0 1px 3px rgba(0, 0, 0, 0.3)',
  shadowHover: '0 4px 12px rgba(6, 182, 212, 0.25)',
};
